pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller.go

```diff
@@ -176,6 +178,22 @@ func (r *NSXServiceAccountReconciler) GarbageCollector(cancel chan bool, timeout
 		}
 		gcSuccessCount, gcErrorCount := r.garbageCollector(nsxServiceAccountUIDSet, nsxServiceAccountList)
 		log.V(1).Info("gc collects NSXServiceAccount CR", "success", gcSuccessCount, "error", gcErrorCount)
+		// Validate ca at first time
+		// Validate client cert every GCValidationInterval
+		if count == 0 {
+			for _, nsxServiceAccount := range nsxServiceAccountList.Items {
+				if nsxserviceaccount.IsNSXServiceAccountRealized(&nsxServiceAccount.Status) {
+					if err := r.Service.ValidateAndUpdateRealizedNSXServiceAccount(context.TODO(), &nsxServiceAccount, ca); err != nil {
```

Comment:
GarbageCollector runs in a different Goroutine than Manager. Do we need to introduce lock between ValidateAndUpdateRealizedNSXServiceAccount and other methods of NSXServiceAccountService?

Reply:
Currently no. I'll add more comments on this.
The same NSXServiceAccount won't be processed by different goroutine.
The PI/CCP cache is thread safe.

Reply:
I also notice that setupWithManager passes runtime.NumCPU() to MaxConcurrentReconciles. I don't find locks being used in NSXServiceAccountService, do we need any locks?

Reply:
No, this property works as `numWorkers`, we don't have the non-threadSafe resources accessed in parallel.